prusti-encoder/src/encoders/mir_pure.rs

```diff
@@ -364,19 +369,18 @@ impl<'vir, 'enc> Encoder<'vir, 'enc>
                     discr.ty(self.body, self.vcx.tcx),
                 ).unwrap();
 
-                // find earliest join point `join`
-                let join = self.find_join_point(dominators, start, end);
-
-                // walk `start` -> `targets[i]` -> `join` for each target
+                // walk `curr` -> `targets[i]` -> `join` for each target. The
+                // join point is identified by reaching a bb where
+                // `dominators.immediate_dominator(bb) == curr`.
```

Comment:
This describes *how* we find the join point, which I think belongs close to where that fact is relevant.